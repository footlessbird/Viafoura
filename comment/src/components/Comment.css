button {
  background: none;
  border: none;
  padding: 0;
  color: lightgray;
  cursor: pointer;
  font-size: 11px;
}

button:hover {
  background: #eeeeee;
}

button:active {
  background-color: #e0e0e0;
  background-size: 100%;
  transition: background 0s;
}

.comment-container {
  text-align: left;
  display: grid;
  grid-template-columns: 1fr 11fr;
  grid-gap: 10px;
  max-width: 800px;
  padding: 20px 0;
  margin-top: 20px;
  border-top: 1px solid lightgray;
  border-bottom: 1px solid lightgray;
}

.comment-container:hover button {
  color: black;
}

.comment-container .profile-image {
  grid-column: 1;
  grid-row: 1 / 3;
}

.profile-image img {
  object-fit: cover;
  height: 50px;
  width: 50px;
  border-radius: 50%;
}

.comment-container .user-name {
  grid-column: 2;
  grid-row: 1;
}

.comment-container .user-name .title {
  background-color: black;
  padding: 1px 10px;
  color: white;
  line-height: 10px;
  border-radius: 500px;
  font-size: 11px;
  text-transform: uppercase;
}

.comment-container .user-name .posting-time {
  color: grey;
  font-size: 11px;
  text-transform: uppercase;
}

.comment-container .content {
  grid-column: 2;
  grid-row: 2;
}

.comment-container .btn-section {
  grid-column: 2;
  grid-row: 3;
  color: lightgray;
  font-size: 11px;
}

.comment-container:hover .btn-section {
  grid-column: 2;
  grid-row: 3;
  color: black;
}

.reply {
  margin-right: 15px;
}

.replies {
  color: gray;
}

.comment-container:hover .replies {
  color: black;
}

.vote {
  margin-left: 15px;
}

.vote button {
  margin-right: 5px;
}
