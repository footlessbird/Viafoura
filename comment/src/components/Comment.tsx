import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronUp, faChevronDown } from "@fortawesome/free-solid-svg-icons";
import "./Comment.css";

type UserProps = {
  user: {
    userName: string;
    title: string;
    profileImage: string;
    lastPosting: string;
    content: string;
    replies: number;
    upvotes: number;
    downvotes: number;
  }[];
};

function Comment({ user }: UserProps) {
  const {
    userName,
    title,
    profileImage,
    lastPosting,
    content,
    replies,
    upvotes,
    downvotes,
  } = user[0];
  return (
    <>
      <li data-testid="comment-container" className="comment-container">
        <div className="profile-image">
          <img src={profileImage} alt="profile image" />
        </div>
        <div className="user-name">
          <b>{userName}</b> <span className="title">{title}</span>
          <span className="posting-time"> &middot; {lastPosting}</span>
        </div>
        <div className="content">{content}</div>
        <section className="btn-section">
          <button className="reply">REPLY</button>
          <span className="replies">{replies}</span> REPLIES
          <span className="vote">
            <button>
              <FontAwesomeIcon className="updown-icon" icon={faChevronUp} />
              {upvotes}
            </button>
          </span>
          <span className="vote">
            <button>
              <FontAwesomeIcon className="updown-icon" icon={faChevronDown} />
              {downvotes}
            </button>
          </span>
        </section>
      </li>
    </>
  );
}

export default Comment;
