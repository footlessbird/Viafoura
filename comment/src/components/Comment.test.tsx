import React from "react";
import { render } from "@testing-library/react";
import Comment from "./Comment";
import data from "../user.json";

describe("<Comment />", () => {
  it("matches snapshot", () => {
    const { user } = data;
    const utils = render(<Comment user={user} />);
    expect(utils.container).toMatchSnapshot();
  });
  it("checks given props and comment-contianer DOM", () => {
    const { user } = data;
    const utils = render(<Comment user={user} />);
    utils.getByTestId("comment-container");
    utils.getByAltText("profile image");
    utils.getByText("Brad");
    utils.getByText(/German/i);
    utils.getByText("21");
  });
});
