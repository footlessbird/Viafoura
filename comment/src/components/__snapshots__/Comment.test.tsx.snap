// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> matches snapshot 1`] = `
<div>
  <li
    class="comment-container"
    data-testid="comment-container"
  >
    <div
      class="profile-image"
    >
      <img
        alt="profile image"
        src="https://randomuser.me/api/portraits/men/75.jpg"
      />
    </div>
    <div
      class="user-name"
    >
      <b>
        Brad
      </b>
       
      <span
        class="title"
      >
        author
      </span>
      <span
        class="posting-time"
      >
         Â· 
        2 MINUTES AGO
      </span>
    </div>
    <div
      class="content"
    >
      So what the German automaker is likely to focus on today is the bigger picture. This will be the first time we see the Taycan free from any prototype bodywork.
    </div>
    <section
      class="btn-section"
    >
      <button
        class="reply"
      >
        REPLY
      </button>
      <span
        class="replies"
      >
        21
      </span>
       REPLIES
      <span
        class="vote"
      >
        <button>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-up fa-w-14 updown-icon"
            data-icon="chevron-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
              fill="currentColor"
            />
          </svg>
          123
        </button>
      </span>
      <span
        class="vote"
      >
        <button>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-down fa-w-14 updown-icon"
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              fill="currentColor"
            />
          </svg>
          0
        </button>
      </span>
    </section>
  </li>
</div>
`;
